src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProject.cs

```diff
@@ -991,14 +991,18 @@ private OneOrMany<string> GetMappedAnalyzerPaths(string fullPath)
             if (fullPath.LastIndexOf(s_razorSourceGeneratorSdkDirectory, StringComparison.OrdinalIgnoreCase) + s_razorSourceGeneratorSdkDirectory.Length - 1 ==
                 fullPath.LastIndexOf(Path.DirectorySeparatorChar))
             {
-                if (fullPath.EndsWith(s_razorSourceGeneratorMainAssemblyRootedFileName, StringComparison.OrdinalIgnoreCase))
+                var vsixRazorAnalyzers = _vsixAnalyzerProvider.GetAnalyzerReferencesInExtensions();
```

Comment:
Won't this include all analyzer references in all extensions? So just having one other analyzer installed (say, our own code analysis analyzers in the VSIX) would break this check?

Reply:
Fixed